### PR TITLE
feat: reports improvement

### LINKED ISSUE #123: Fix: Error al descargar "Epidemioligía - Vigilancia Epidemiológica"
### Descripción
Se observó el siguiente error al momento de generar el reporte en el entorno productivo de HSI cuando la fecha de inicio es igual o anterior a la fecha 14/02/2024.
Además de esto, los reportes que si se logran generar, con una fecha posterior al 14/02/2024 como fecha de inicio, no devuelven resultados.

### Pasos para replicar el error
1. Ingresar con usuario de administrador.
2. Elegir una institución.
3. Ir al módulo de "Reportes provinciales"
4. Situarse en la pestaña de "Programas"
5. Seleccionar el reporte "Epidemiología - Vigilancia epidemiológica"
6. Definir fecha de inicio:
- anterior o igual al 14/02/2024
7. Definir fecha de fin:
- fecha actual
8. Hacer click en "Generar reporte".

### LINKED ISSUE #125: [ Fix ] Modificación encabezado "Epidemiología - Vigilancia epidemiológica"
## Descripción 
Se pide realizar una actualización en el diseño del encabezado del reporte y eliminar columnas con datos innecesarios.

Para la elaboración de las modificaciones se recomienda ver el siguiente archivo adjunto: 
[epidemiologia-vigilancia-epidemiologica-modelo.xlsx](https://github.com/Historia-Clinica-La-Rioja/historia_clinica_LR/files/15296889/epidemiologia-vigilancia-epidemiologica-modelo.xlsx)

### Enlace a Taiga
LINK: https://tree.taiga.io/project/rlrama-tablero-de-scrum-ministerio-de-salud-publica-de-la-rioja/us/72

## Resultado esperado
image

### LINKED ISSUE #126: [ Fix ] Modificación de encabezado "Epidemiología – Notificación colectiva"
## Descripción 
Se pide realizar una actualización en el diseño del encabezado del reporte y eliminar columnas con datos innecesarios.

Para la elaboración de las modificaciones se recomienda ver el siguiente archivo adjunto: 
[epidemiologia-notificacion-colectiva-modelo.xlsx](https://github.com/Historia-Clinica-La-Rioja/historia_clinica_LR/files/15296968/epidemiologia-notificacion-colectiva-modelo.xlsx)

### Enlace a Taiga
https://tree.taiga.io/project/rlrama-tablero-de-scrum-ministerio-de-salud-publica-de-la-rioja/us/73
## Resultado esperado
image

### LINKED ISSUE #127: [ Fix ] Modificación de encabezado "Recupero - Obras Sociales"
## Descripción
Se pide realizar una actualización en el diseño del encabezado del informe y eliminar columnas con datos innecesarios.

Para la elaboración de las modificaciones se recomienda ver el siguiente archivo adjunto:
[reporte-de-recupero-obras-sociales-modelo.xlsx](https://github.com/Historia-Clinica-La-Rioja/historia_clinica_LR/files/15297023/reporte-de-recupero-obras-sociales-modelo.xlsx)

### Enlace a Taiga
https://tree.taiga.io/project/rlrama-tablero-de-scrum-ministerio-de-salud-publica-de-la-rioja/us/74
## Resultado esperado
image

### LINKED ISSUE #128: [ Fix ] Modificación de encabezado "Recupero Odontológico - Obras Sociales"
Descripción
Se pide realizar una actualización en el diseño del encabezado del informe y eliminar columnas con datos innecesarios.

Para la elaboración de las modificaciones se recomienda ver el siguiente archivo adjunto:
[reporte-de-recupero-odontologico-obras-sociales-modelo (1).xlsx](https://github.com/Historia-Clinica-La-Rioja/historia_clinica_LR/files/15297154/reporte-de-recupero-odontologico-obras-sociales-modelo.1.xlsx)

Enlace a Taiga
https://tree.taiga.io/project/rlrama-tablero-de-scrum-ministerio-de-salud-publica-de-la-rioja/us/77

## Resultado esperado
image

### LINKED ISSUE #129: [ Fix ] Modificación de encabezado "Programa Sumar"
## Descripción
Se pide realizar una actualización en el diseño del encabezado del informe y eliminar columnas con datos innecesarios.

Para la elaboración de las modificaciones se recomienda ver el siguiente archivo adjunto:
[reporte-del-programa-sumar-modelo.xlsx](https://github.com/Historia-Clinica-La-Rioja/historia_clinica_LR/files/15297227/reporte-del-programa-sumar-modelo.xlsx)

### Enlace a Taiga
https://tree.taiga.io/project/rlrama-tablero-de-scrum-ministerio-de-salud-publica-de-la-rioja/us/75

### LINKED ISSUE #130: [ Fix ] Modificción de encabezado "Reporte de Sumar – Odontológico"
## Descripción
Se pide realizar una actualización en el diseño del encabezado del informe y eliminar columnas con datos innecesarios.

Para la elaboración de las modificaciones se recomienda ver el siguiente archivo adjunto:
[reporte-de-sumar-odontologico-modelo.xlsx](https://github.com/Historia-Clinica-La-Rioja/historia_clinica_LR/files/15301023/reporte-de-sumar-odontologico-modelo.xlsx)

### Enlace a Taiga
https://tree.taiga.io/project/rlrama-tablero-de-scrum-ministerio-de-salud-publica-de-la-rioja/us/76

### PR DESCRIPTION
**Pull request by RLRama**
## Description

This PR introduces:
- new headers for all the **program reports**,
- **update queries** for the two 'Programa Sumar - ...' and the two 'Recupero - ...' reports (**1**, **4**, **5**, **6**)
- **extraction** of the method used for calculating the BMI in the 'Sumar General' report
- code refactoring (reduction of boilerplate, creation of common utilities), and
- fixes error when downloading report **2** (a date formatter that didn't work properly).

The affected reports are the following:

1. [Programa Sumar - General - 12.12.2020 - 12.12.2024.xls](https://github.com/user-attachments/files/15823327/Programa.Sumar.-.General.-.12.12.2020.-.12.12.2024.xls)
2. [Epidemiología - Vigilancia epidemiológica - 12.12.2020 - 12.12.2024.xls](https://github.com/user-attachments/files/15823329/Epidemiologia.-.Vigilancia.epidemiologica.-.12.12.2020.-.12.12.2024.xls)
3. [Epidemiología - Notificación colectiva - 12.12.2020 - 12.12.2024.xls](https://github.com/user-attachments/files/15823330/Epidemiologia.-.Notificacion.colectiva.-.12.12.2020.-.12.12.2024.xls)
4. [Recupero - Obras sociales - 12.12.2020 - 12.12.2024.xls](https://github.com/user-attachments/files/15823331/Recupero.-.Obras.sociales.-.12.12.2020.-.12.12.2024.xls)
5. [Recupero - Odontológico - 12.12.2020 - 12.12.2024.xls](https://github.com/user-attachments/files/15823332/Recupero.-.Odontologico.-.12.12.2020.-.12.12.2024.xls)
6. [Programa Sumar - Odontológico - 12.12.2020 - 12.12.2024.xls](https://github.com/user-attachments/files/15823333/Programa.Sumar.-.Odontologico.-.12.12.2020.-.12.12.2024.xls)

**NOTES:**
test date range is arbitrary, and the reports were generated before the query updates (results may differ).

## Changes

This reunites all the methods shared between program reports and the upcoming reports refactor, so boilerplate is reduced. The controllers access routes are left intact, so front-end report generation should work correctly with no required changes.

## Related issues

- Closes #123
- Closes #125
- Closes #126 
- Closes #127
- Closes #128
- Closes #129
- Closes #130

### CODE REVIEW

**Comment by RLRama** [comment]
**Last @cquevedox PR closed in person?**

File: back-end/hospital-api/src/main/java/net/pladema/provincialreports/programreports/controller/ProgramReportsController.java [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+			logger.info("Successfully wrote 'recupero - obras sociales' report for institutionId: {}", institutionId);
+		} catch (Exception e) {
+			logger.error("Error generating 'recupero - obras sociales' report for institutionId: {}", institutionId, e);
+			throw e;
+		}
+
+	}
+
+
+////////////////////////////////////////
```
Comment (cquevedox) [inline back-end/hospital-api/src/main/java/net/pladema/provincialreports/programreports/controller/ProgramReportsController.java:162]: ??

File: back-end/hospital-api/src/main/java/net/pladema/provincialreports/programreports/controller/ProgramReportsController.java [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+			outputStream.flush();
+			response.flushBuffer();
+
+			logger.info("Successfully wrote 'sumar general' report for institutionId: {}", institutionId);
+		} catch (Exception e) {
+			logger.error("Error generating 'sumar general' report for institutionId: {}", institutionId, e);
+			throw e;
+		}
+
+	}////////////////////////////////////////////////
```
Comment (cquevedox) [inline back-end/hospital-api/src/main/java/net/pladema/provincialreports/programreports/controller/ProgramReportsController.java:213]: ?

File: back-end/hospital-api/src/main/java/net/pladema/provincialreports/programreports/repository/ProgramReportQueryFactory.java [hunk 3]
```diff
@@ -82,6 +82,7 @@ public List<OdontologicalConsultationDetail> queryRecuperoOdontologico(Integer i
 
 	}
 
+
```
Comment (cquevedox) [inline back-end/hospital-api/src/main/java/net/pladema/provincialreports/programreports/repository/ProgramReportQueryFactory.java:85]: ?

**Review by cquevedox — APPROVED** [review]

**Pull request merged**